### PR TITLE
Jethro merge

### PR DESCRIPTION
**Pull request by aikidokatech**
These changes are for building the xenclient-oe repo against the current OE Jethro (2.0) release.  Make sure to use a new or updated build host/container.  See  https://openxt.atlassian.net/wiki/display/BS/Jessie+systemd+container for instructions on creating a new Jessie systemd container.

### CODE REVIEW

File: setup_build-next.sh [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+#######################################################################
+update_git_repo() {
+	local path="$1"
+	local repo="$2"
+
+	echo "Updating local copy of $repo..."
+	cd $path
+	set +e
+	#  Simple pull only works if fast forware is possible.
+	git pull || die "Update of git repo failed: $path  Please resolve manually."
```
Comment (jean-edouard) [inline setup_build-next.sh]: This doesn't make sense for clones like meta-java.
Also, "die" is not defined.

**Comment by rossphilipson** [comment]
I agree with @jean-edouard about this. I don't see why we would do pulls on the OE repos after the setupoe step has been run the first time. It seems it would be better for the setupoe step to see it had run and just exit.

File: setup_build-next.sh (hunk 1 above)
Comment (aikidokatech) [inline setup_build-next.sh]: This section was not yet finished or vetted as it was new functionality.

**Comment by aikidokatech** [comment]
An OE branch is not frozen once created.  It receives bug fixes and
security updates.  The goal of this work is to simplify the updating of the
repos to pull in the latest patches.  If removed one will need to manually
update the repos.

**Comment by jean-edouard** [comment]
@aikidokatech for some repos we checkout a specific commit ID, which pretty much means we don't want auto-updates.

Now regarding the OE branch, that's a very important topic, as including the tip of a moving branch breaks build reproducibility.
Now maybe it's ok for master follow the tip of the OE jethro branch, and stable-X to build against a specific commit ID? master builds wouldn't be reproducible, but stable-X ones would...
Maybe that's a discussion for the mailing list?

**Comment by aikidokatech** [comment]
Correct, meta-java will be an exception for now.  There are no other repos
that use a specific hash.

I disagree about the other repos.  The updates they receive should
typically not break a build as the branches are intended to be stable once
released.  Discussing this on the list would get better visibility.

**Pull request merged**